Menus/ScenarioActivityConfigGUI.cpp

```diff
@@ -0,0 +1,435 @@
+			if (m_SelectedActivity->TeamActive(team)) {
+				teamIcon = m_SelectedActivity->GetTeamIcon(team);
+				if (!teamIcon) { teamIcon = dynamic_cast<const Icon *>(g_PresetMan.GetEntityPreset("Icon", "Team " + std::to_string(team + 1) + " Default")); }
+				m_TeamNameLabels.at(team)->SetText(m_SelectedActivity->GetTeamName(team));
+			} else {
+				teamIcon = dynamic_cast<const Icon *>(g_PresetMan.GetEntityPreset("Icon", "Locked Team"));
+				m_TeamNameLabels.at(team)->SetText("Unavailable");
+			}
+			if (teamIcon) { m_TeamIconBoxes.at(team)->SetDrawImage(new AllegroBitmap(teamIcon->GetBitmaps32()[0])); }
+
+			m_TeamTechComboBoxes.at(team)->SetVisible(m_SelectedActivity->TeamActive(team));
+			m_TeamAISkillSliders.at(team)->SetVisible(m_SelectedActivity->TeamActive(team));
+			m_TeamAISkillLabels.at(team)->SetVisible(m_SelectedActivity->TeamActive(team));
+		}
+		if (const Icon *disabledTeamIcon = dynamic_cast<const Icon *>(g_PresetMan.GetEntityPreset("Icon", "Disabled Team"))) { m_TeamIconBoxes.at(TeamRows::DisabledTeam)->SetDrawImage(new AllegroBitmap(disabledTeamIcon->GetBitmaps32()[0])); }
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioActivityConfigGUI::StartGame() {
+		GameActivity *gameActivity = dynamic_cast<GameActivity *>(m_SelectedActivity->Clone());
+
+		gameActivity->SetDifficulty(m_ActivityDifficultySlider->GetValue());
+		gameActivity->SetStartingGold((m_StartingGoldSlider->GetValue() == m_StartingGoldSlider->GetMaximum()) ? 1000000000 : m_StartingGoldSlider->GetValue() - (m_StartingGoldSlider->GetValue() % 500));
+
+		gameActivity->SetRequireClearPathToOrbit(m_RequireClearPathToOrbitCheckbox->GetCheck());
+		gameActivity->SetFogOfWarEnabled(m_FogOfWarCheckbox->GetCheck());
+		g_SceneMan.SetSceneToLoad(m_SelectedScene, true, m_DeployUnitsCheckbox->GetCheck());
+
+		gameActivity->ClearPlayers(false);
+		for (int player = Players::PlayerOne; player < PlayerColumns::PlayerColumnCount; ++player) {
+			for (int team = Activity::Teams::TeamOne; team < Activity::Teams::MaxTeamCount; ++team) {
+				if (m_PlayerBoxes.at(player).at(team)->GetDrawType() == GUICollectionBox::Image) {
+					if (player == PlayerColumns::PlayerCPU) {
+						gameActivity->SetCPUTeam(team);
+					} else {
+						gameActivity->AddPlayer(player, true, team, 0);
+					}
+					break;
+				}
+			}
+		}
+
+		for (int team = Activity::Teams::TeamOne; team < Activity::Teams::MaxTeamCount; ++team) {
+			if (const GUIListPanel::Item *techItem = m_TeamTechComboBoxes.at(team)->GetSelectedItem()) {
+				if (techItem->m_ExtraIndex == -2) {
+					gameActivity->SetTeamTech(team, "-All-");
+				} else if (techItem->m_ExtraIndex == -1) {
+					m_TeamTechComboBoxes.at(team)->SetSelectedIndex(RandomNum<int>(2, m_TeamTechComboBoxes.at(team)->GetListPanel()->GetItemList()->size() - 1));
+					gameActivity->SetTeamTech(team, g_PresetMan.GetDataModuleName(techItem->m_ExtraIndex));
```

Comment:
Does this work without setting `techItem = m_TeamTechComboBoxes.at(team)->GetSelectedItem()` again?